pkg/network/ebpf/c/protocols/http2/telemetry.h

```diff
@@ -0,0 +1,60 @@
+#ifndef __HTTP2_TELEMETRY_H
+#define __HTTP2_TELEMETRY_H
+
+#include "ktypes.h"
+#include "maps-defs.h"
+
+enum telemetry_counter
+{
+    end_of_stream_eos,
+    end_of_stream_rst,
+    str_len_greater_then_frame_loc,
+    str_len_too_big_mid,
+    str_len_too_big_large,
+    request_seen,
+    response_seen,
+    frame_remainder,
+    max_frames_in_packet,
+};
+
+static __always_inline void increment_telemetry_count(enum telemetry_counter counter_name) {
```

Comment:
I'm concerned this code adds a lot of complexity.
Instead you should be calling `bpf_map_lookup_elem(&http2_telemetry, &key)` once, and pass the map pointer to the methods.